subsquid/schema.graphql

```diff
@@ -96,10 +93,100 @@ type VestingSchedule @entity {
   from: String!
   "chain event ID"
   eventId: String!
-  "{accoundId}-{assetId}"
-  scheduleId: String!
+  "Vesting schedule ID from chain"
+  scheduleId: BigInt!
   "'To' account for the vesting schedule"
   to: String!
+  "Asset ID"
+  asset: BigInt!
   "Vesting schedule"
   schedule: Schedule!
+  "Initial locked amount"
+  totalAmount: BigInt!
+  "True if the schedule has been fully claimed"
+  fullyClaimed: Boolean!
+}
+
+enum PicassoTransactionType {
+  BALANCES_TRANSFER,
+  BALANCES_DEPOSIT,
+  BALANCES_WITHDRAW,
+  STAKING_REWARDS_REWARD_POOL_CREATED,
+  STAKING_REWARDS_STAKED,
+  STAKING_REWARDS_STAKE_AMOUNT_EXTENDED,
+  STAKING_REWARDS_UNSTAKE,
+  STAKING_REWARDS_SPLIT_POSITION,
+  BONDED_FINANCE_NEW_OFFER,
+  BONDED_FINANCE_NEW_BOND,
+  BONDED_FINANCE_OFFER_CANCELLED,
+  VESTING_SCHEDULES_VESTING_SCHEDULE_ADDED,
+  VESTING_SCHEDULES_CLAIMED,
+  VESTING_SCHEDULES_VESTING_SCHEDULES_UPDATED,
+}
+
+type PicassoTransaction @entity {
+  id: ID!
+  "ID of the event that was used to derive this transaction"
+  eventId: String!
+  "ID of account that executed transaction"
+  accountId: String!
+  "Type of transaction"
+  transactionType: PicassoTransactionType!
+  "Block in which transaction was registered"
+  blockNumber: BigInt!
+  "Timestamp of the block in which this transaction was registered"
+  timestamp: BigInt!
+}
+
+type Activity @entity {
+  id: ID!
+  "ID of the event associated with this activity"
+  eventId: String!
+  "ID of the transaction associated with this activity"
+  transactionId: String!
+  "ID of the active account"
+  accountId: String!
+  "Timestamp of the block in which this activity occurred"
+  timestamp: BigInt!
+}
+
+type Asset @entity {
+  "ID of the asset in Picasso"
+  id: ID!
+  "ID of the event that added the last price"
+  eventId: String!
+  price: BigInt!
+  historicalPrices: [HistoricalAssetPrice!] @derivedFrom(field: "asset")
+}
+
+type HistoricalAssetPrice @entity {
+  id: ID!
+  "ID of the event associated with this activity"
+  eventId: String!
+  "ID of the asset"
+  asset: Asset!
+  price: BigInt!
+  timestamp: BigInt!
+}
+
+type PicassoStakingPosition @entity {
```

Comment:
We can model a single staking position and have them distinguished by the staked assetId instead of creating a data structure specific to one asset type

Reply:
Yeah, the thing is I want to store them in USD so that we have a good representation of historical data. I will think of a better way to make it more flexible.

Reply:
I also think that `USD` is mera here. it would be nice to add a currency field to this to declare the price in that currency.